File: backend/app/Models/WeatherAdvisory.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class WeatherAdvisory extends Model
{
    protected $table = 'weather_advisories';

    protected $fillable = [
        'school_id', 'kind', 'severity', 'message', 'message_ar',
        'active', 'starts_at', 'ends_at',
    ];

    protected function casts(): array
    {
        return [
            'active' => 'boolean',
            'starts_at' => 'datetime',
            'ends_at' => 'datetime',
        ];
    }

    public function school(): BelongsTo
    {
        return $this->belongsTo(School::class);
    }
}

File: flutter_app/lib/data/models/weather_advisory.dart
/// A weather/AQI advisory, matching `WeatherAdvisoryResource` from the API.
class WeatherAdvisory {
  const WeatherAdvisory({
    required this.id,
    this.schoolId,
    required this.kind,
    this.severity,
    required this.message,
    this.messageAr,
    this.active = false,
    this.startsAt,
    this.endsAt,
    this.createdAt,
  });

  final int id;
  final int? schoolId;
  final String kind; // haboob | aqi-dust | heat | flooding | other | ...
  final String? severity;
  final String message;
  final String? messageAr;
  final bool active;
  final DateTime? startsAt;
  final DateTime? endsAt;
  final DateTime? createdAt;

  factory WeatherAdvisory.fromJson(Map<String, dynamic> j) => WeatherAdvisory(
        id: (j['id'] as num).toInt(),
        schoolId: (j['school_id'] as num?)?.toInt(),
        kind: j['kind'] as String? ?? '',
        severity: j['severity'] as String?,
        message: j['message'] as String? ?? '',
        messageAr: j['message_ar'] as String?,
        active: j['active'] as bool? ?? false,
        startsAt: _date(j['starts_at']),
        endsAt: _date(j['ends_at']),
        createdAt: _date(j['created_at']),
      );

  static DateTime? _date(Object? v) =>
      v is String && v.isNotEmpty ? DateTime.tryParse(v) : null;
}
